src/terminal/parser/stateMachine.cpp

```diff
@@ -1415,40 +1305,41 @@ void StateMachine::ProcessString(const wchar_t* const rgwch, const size_t cch)
         // means we'll make sure to call `_ActionEscDispatch('[')`., which will
         // properly decode the string as <kbd>alt+[</kbd>.
 
-        if (_pEngine->FlushAtEndOfString())
+        if (_engine->FlushAtEndOfString())
         {
             // Reset our state, and put all but the last char in again.
             ResetState();
             // Chars to flush are [pwchSequenceStart, pwchCurr)
-            const wchar_t* pwch = _pwchSequenceStart;
-            for (; pwch < _pwchCurr - 1; pwch++)
+            auto wchIter = _run.cbegin();
+            while (wchIter < _run.cend() - 1)
             {
-                ProcessCharacter(*pwch);
+                ProcessCharacter(*wchIter);
+                wchIter++;

```

Comment:
this is a fancy way to write `for(const auto wch : _run)`, right?

Reply:
oh, it's the `-1` that gets it innit

Reply:
Except it's more like `for (const auto wch : _run.substr(0, _run.size() - 1))` as it leaves the last one off.